aodncore/util/misc.py

```diff
@@ -82,6 +83,19 @@ def ensure_pattern(o):
     return re.compile(o)
 
 
+def ensure_pattern_list(o):
```

Comment:
I think this makes `validate_regexes` obsolete, no? In fact, `validate_regex` itself is now only called from `ensure_pattern`, so maybe could be combined?

Reply:
It is obsolete in the sense that nothing is using it right now, but validate_* and ensure_* have a different purpose, with validate being a lower level concept, so I think it makes sense to keep them separate.

**validate_thing** doesn't return anything, and causes a "do nothing if thing is valid, exception if invalid". **ensure_thing** is all about *guaranteeing* to return a **thing**.

Some functions still use/need the basic "validate only" behaviour of validate_* , but some functions fit the model of "validate + coerce" of ensure_* better.

Reply:
See latest commit. "regex" is used in function/method names to be unambiguous, and "patterns" are used for parameter names.

Reply:
Although there is now a mismatch with the handler parameters which use regex. Argh! That said, I think out in JSON land, it's reasonable to be explicit to users that the patterns are regexes, and not glob patterns.